### PR TITLE
Error code formatting ( Will work later )

### PR DESCRIPTION
**Pull request by ShrijanL**
- Created custom exception class.
- Formatted ERRORCODES.md file.
- Used CustomAPIError class to raises and catches the errors.

### CODE REVIEW

File: django_generic_api/services.py [hunk 1]
```diff
@@ -56,7 +57,9 @@ def get_model_by_name(model_name):
             model = app_config.models.get(model_name.lower())
             if model:
                 return model
-    raise ValueError
+    raise CustomAPIError(
+        "Model not found", "DGA-S001", status.HTTP_404_NOT_FOUND
```
Comment (nareshelequin) [inline django_generic_api/services.py:61]: Use, kwargs instead args.

File: django_generic_api/utils.py [hunk 2]
```diff
@@ -77,16 +88,18 @@ def is_fields_exist(model, fields):
                 related_model_meta = getattr(fk.related_model, "_meta")
                 related_model_meta.get_field(related_field)
             except FieldDoesNotExist:
```
Comment (nareshelequin) [inline django_generic_api/utils.py:90]: How do you handle? if something else exception raised?

File: django_generic_api/views.py [hunk 3]
```diff
@@ -366,12 +456,22 @@ def get(self, request, encoded_token, *args, **kwargs):
                 status=status.HTTP_201_CREATED,
             )
         except User.DoesNotExist:
+            custom_error = CustomAPIError(
```
Comment (nareshelequin) [inline django_generic_api/views.py:459]: Seems `CustomAPIError` class is not used as expected way. Try using expected way.

**Review by nareshelequin — CHANGES_REQUESTED** [review]

File: django_generic_api/utils.py (hunk 2 above)
Comment (ShrijanL) [reply django_generic_api/utils.py:90]: Added Execption case.

**Comment by ShrijanL** [comment]
will work later on this issue
#38

**Pull request closed without merging**